source-kafka/src/kafka.rs

```diff
@@ -50,16 +76,33 @@ pub fn consumer_from_config(configuration: &Configuration) -> Result<BaseConsume
 
     config.set("security.protocol", configuration.security_protocol());
 
-    if let Some(ref auth) = configuration.authentication {
-        config.set("sasl.mechanism", auth.mechanism.to_string());
-        config.set("sasl.username", &auth.username);
-        config.set("sasl.password", &auth.password);
+    let ctx = FlowConsumerContext { auth: configuration.credentials.clone() };
+
+    if let Some(Credentials::UserPassword { mechanism, username, password }) = &configuration.credentials {
+        config.set("sasl.mechanism", mechanism.to_string());
+        config.set("sasl.username", username);
+        config.set("sasl.password", password);
+    } else if let Some(Credentials::AWS { .. }) = &configuration.credentials {
+        config.set("sasl.mechanism", "OAUTHBEARER");
+
+        if configuration.security_protocol() != "SASL_SSL" {
+            return Err(eyre::eyre!("must use tls=system_certificates for AWS").into())
+        }
+    }
+
+    let consumer: BaseConsumer<FlowConsumerContext> = config.create_with_context(ctx).map_err(Error::Config)?;
+
+    if let Some(Credentials::AWS { .. }) = &configuration.credentials {
+        // In order to generate an initial OAuth Bearer token to be used by the consumer
+        // we need to call poll once.
+        // See https://docs.confluent.io/platform/current/clients/librdkafka/html/classRdKafka_1_1OAuthBearerTokenRefreshCb.html
+        let _ = consumer.poll(KAFKA_TIMEOUT);
```

Comment:
This seems to be ignoring the result, which could contain an error.

Reply:
@psFried this is expected to fail as there is no topic assignment yet, this is only done so that rdkafka will generate an oauth token and assign it to the consumer.

If there is any _real error_ in connecting, we will see it subsequently in `test_connection`

Reply:
Added a comment explaining this, too